File: 74.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int n = matrix.size() * matrix[0].size();
        int rit = n-1;
        int left= 0;
        int m = 0;
        int i = (m / matrix[0].size());
        int j = (m % matrix[0].size());
        while(left <= rit){
            m = left + (rit - left)/2;
            i = (m / matrix[0].size());
            j = (m % matrix[0].size());
            if(matrix[i][j] == target){
                return true;
            }else if(matrix[i][j] > target){
                rit = m-1;
            }else{
                left = m + 1;
            }
        }
        return false;
    }
};

File: Python/74.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        n = len(matrix) * len(matrix[0])
        l, r = 0, n-1
        while l <= r:
            m = l + ((r - l) // 2)
            i = m // len(matrix[0])
            j = m % len(matrix[0])
            if matrix[i][j] > target:
                r = m-1
            elif matrix[i][j] < target:
                l = m+1
            else:
                return True
        return False
